src/main/java/com/capitalone/dashboard/webhook/github/GitHubPullRequestV3.java

```diff
@@ -510,6 +511,22 @@ protected void updateMatchingCommitsInDb(Commit commit, GitRequest pull) {
         LOG.debug("Time to make commitRepository call = "+(end-start));
     }
 
+    // Add pull number to merge commits for the PR if they don't have one, in case of rebase merge or squash merge
+    private void updateCommitsWithPullNumber(GitRequest pull) {
+        long start = System.currentTimeMillis();
+        List<Commit> commitsInDb
+                = commitRepository.findByScmRevisionNumber(pull.getScmRevisionNumber());
+
```

Comment:
I feel its not appropriate to run foreach on a empty collection just split it to 2 segments. `if (CollectionUtils.isEmpty(commitsInDb) return;` and `commitsInDb.forEach(commitInDb -> {
                    commitInDb.setPullNumber(pull.getNumber());
                    commitRepository.save(commitInDb);
                });`